### PR TITLE
Fix errors setting up levels during rebases and bisects

### PR DESCRIPTION
**Pull request by benthayer**
During a rebase conflict or a bisect, the head branch would be considered checked out. Since a checked out branch couldn't be deleted, we had trouble setting up new levels during resets/loads/re-inits. This should fix that.

#290 Needs this

### CODE REVIEW

**Comment by benthayer** [comment]
@sahansk2 Would you mind taking a look at this?

File: gitgud/test_operations.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        # This will happen every time
+        pass
+
+
+def setup_bisect():
+    file_operator = get_operator()
+    file_operator.repo.git.bisect('start')
+
+
+def test_reset_during_rebase_conflict(gg):
```
Comment (sahansk2) [inline gitgud/test_operations.py:33]: Minor nitpick, I'd change the testing to use fixtures instead, but that's not the point of this PR. So, feel free to keep it as it is, if you want.

**Review by sahansk2 — APPROVED** [review]
Yep, looks good! Just added a [minor] comment about testing.

File: gitgud/test_operations.py (hunk 1 above)
Comment (benthayer) [reply gitgud/test_operations.py]: We got a pytest

**Pull request merged**